### PR TITLE
blueprints: Adding a BP000 - v2

### PR DESCRIPTION
**Pull request by beraldoleal**
After discussions, this is marked as 'approved'.

Signed-off-by: Beraldo Leal <bleal@redhat.com>

Changes from v1:

 - Moved from 'draft' to 'approved'
 - A lot of changes suggested by @pevogam

### CODE REVIEW

**Comment by beraldoleal** [comment]
@clebergnu, @willianrampazzo @richtja, @pevogam:

I incorporated here a lot of changes proposed by you guys. Please let me know if this is ready for merge.

**Comment by pevogam** [comment]
Hi @beraldoleal, I hope we have some time to look into this and not rush it. Or we will end up with unbaked "internal" PR-s again similarly to commit messages ;)

**Comment by beraldoleal** [comment]
> Hi @beraldoleal, I hope we have some time to look into this and not rush it. Or we will end up with unbaked "internal" PR-s again similarly to commit messages ;)

I'm sending a v2 and asking for review, I'm not rushing it. Take your time.

**Comment by pevogam** [comment]
> > Hi @beraldoleal, I hope we have some time to look into this and not rush it. Or we will end up with unbaked "internal" PR-s again similarly to commit messages ;)
> 
> I'm sending a v2 and asking for review, I'm not rushing it.

Sorry, I was referring to your comment https://github.com/avocado-framework/avocado/pull/4262#issuecomment-727967041 and the "so we can move forward with this ASAP".

> Take your time.

Thanks!

**Review by richtja — APPROVED** [review]
LGTM, thanks

**Review by willianrampazzo — APPROVED** [review]

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:30]: I generally think Avocado is far from a small project so definitely useful to have the blueprints here. They probably also arose naturally from the complexity and impactful decisions involved in each architectural change.
(just a small comment, not requesting to change anything)

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:40]: "socialize" is a bit repetitive here, perhaps "so we could better communicate our decisions to future readers"?

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:42]: It is much clearer now how the two relate, thanks for clearing up all of this!

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:50]: I would probably keep this in the same style as the first item, i.e. 'to solve the problem? (your "solution");'

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:59]: would write "RFCs, as we use them today, are just..."

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:76]: Especially if we are resolving large issues through multiple PRs- blueprints will for sure be indispensable there.

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:105]: ...we *are* fixing ... without thinking *about* the big picture...

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:106]: I guess this also implies we don't have any requirements about the size of the blueprints and a blueprint might be fairly small and to the point, correct?

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:133]: there is no comma here

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:156]: I didn't get the last sentence: Like epic issues, epic blueprints don't need to provide details that the sub-blueprints would clarify?

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:218]: technical

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:252]: Below you can find a brief description...

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:276]: "feature/solution" is repeated here

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:290]: Something that remains slightly unclear here: should the issues be just listed as #numbers or as links to GitHub?

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:300]: 100% compliant with...

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [inline docs/source/blueprints/BP000.rst:336]: Is this meant as a document called TODO which is the reference or as "references are still to be done"?

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [reply docs/source/blueprints/BP000.rst:133]: but there are some other commas lying around in strange places so feel free to ignore that

**Review by pevogam — COMMENTED** [review]
Hi @beraldoleal thank you for your patience! I left a few tiny comments, the blueprint is almost done in my eyes.

File: docs/source/blueprints/BP000.rst
Comment (beraldoleal) [reply docs/source/blueprints/BP000.rst:40]: In this section, I'm using the word "socialize" twice. On in each paragraph. Yes, It was my intention to reinforce this idea.

I would like to keep it.

File: docs/source/blueprints/BP000.rst
Comment (beraldoleal) [reply docs/source/blueprints/BP000.rst:106]: Yes.

File: docs/source/blueprints/BP000.rst
Comment (beraldoleal) [reply docs/source/blueprints/BP000.rst:156]: Changing to this:

> Like an epic issue, epic blueprints don't need to detail "how" (or provide details) that the sub-blueprints could have.

File: docs/source/blueprints/BP000.rst
Comment (beraldoleal) [reply docs/source/blueprints/BP000.rst:290]: On the "_Related Issues_" description we have: "Here, **you should mention Github links for both**: a) current open issues that are blocking while waiting for your BP and b) all open issues that will render this BP as “implemented” when closed."

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [reply docs/source/blueprints/BP000.rst:156]: I guess in this case it should be: "Like an epic issue, epic blueprints don't need to detail the "how" that the sub-blueprints could have.", i.e. just an extra "the" so that it is understandable we refer to the "how" above.

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [reply docs/source/blueprints/BP000.rst:106]: Might be worth mentioning it but I guess it is clear to everyone.

File: docs/source/blueprints/BP000.rst
Comment (beraldoleal) [reply docs/source/blueprints/BP000.rst:336]: Here is to add references for your BP. But in this case, we don't have, so I will remove this word. THanks

File: docs/source/blueprints/BP000.rst
Comment (pevogam) [reply docs/source/blueprints/BP000.rst:40]: Alright , it seems over-repetitive for me but your call, I guess you have a goal behind this.

**Comment by beraldoleal** [comment]
Hi @pevogam, thanks again for your detailed review. I have incorporated most of your comments. Please, let me know if you are ok with the new version.

**Review by pevogam — APPROVED** [review]
Hi @beraldoleal, three passes should be enough. The first meta-blueprint is good to go!

**Comment by beraldoleal** [comment]
> Hi @beraldoleal, three passes should be enough. The first meta-blueprint is good to go!

Yes, but since this is a very important one, I wait for @clebergnu's review.

**Comment by pevogam** [comment]
> > Hi @beraldoleal, three passes should be enough. The first meta-blueprint is good to go!
> 
> Yes, but since this is a very important one, I wait for @clebergnu's review.

One more reason it was good that we spend more energy on this one. And thanks for your continued patience.

**Comment by beraldoleal** [comment]
@clebergnu ?

**Comment by clebergnu** [comment]
> @clebergnu ?

Let's get this merged!

**Pull request merged**
